src/cli/one_helper/onebackupjob_helper.rb

```diff
@@ -0,0 +1,286 @@
+
+        perform_actions(ids, options, message) do |bj|
+            rc = bj.sched_action_add(tmp_str)
+
+            if OpenNebula.is_error?(rc)
+                STDERR.puts rc.message
+                exit(-1)
+            end
+        end
+    end
+
+    # Update schedule action
+    #
+    # @param id        [Integer] BackupJob ID
+    # @param action_id [Integer] Sched action ID
+    # @param file      [String]  File path with update content
+    # @param options
+    def update_schedule_action(id, action_id, file, options)
+        perform_action(id, options, 'Sched action updated') do |bj|
+            rc = bj.info
+
+            if OpenNebula.is_error?(rc)
+                STDERR.puts "Error #{rc.message}"
+                exit(-1)
+            end
+
+            xpath = "TEMPLATE/SCHED_ACTION[ID=#{action_id}]"
+
+            unless bj.retrieve_elements(xpath)
+                STDERR.puts "Sched action #{action_id} not found"
+                exit(-1)
+            end
+
+            # Get user information
+            if file
+                str = File.read(file)
+            elsif !(stdin = OpenNebulaHelper.read_stdin).empty?
+                str = stdin
+            else
+                str = OpenNebulaHelper.update_template(id, bj, nil, xpath)
+            end
+
+            # Add the modified sched action
+            tmp_str = "\nSCHED_ACTION = ["
+            tmp_str << str.split("\n").join(',')
+            tmp_str << ']'
+
+            rc = bj.sched_action_update(action_id, tmp_str)
+
+            if OpenNebula.is_error?(rc)
+                STDERR.puts "Error updating: #{rc.message}"
+                exit(-1)
+            end
+        end
+    end
+
+    def self.create_backupjob_template(options)
+        template_options = TEMPLATE_OPTIONS.map do |o|
+            o[:name].to_sym
+        end
+
+        template_options << :name
+
+        t = ''
+        template_options.each do |n|
+            t << "#{n.to_s.upcase}=\"#{options[n]}\"\n" if options[n]
+        end
+
+        t
+    end
+
+    private
+
+    def factory(id = nil)
+        if id
+            OpenNebula::BackupJob.new_with_id(id, @client)
+        else
+            xml=OpenNebula::BackupJob.build_xml
+            OpenNebula::BackupJob.new(xml, @client)
+        end
+    end
+
+    def factory_pool(user_flag = -2)
+        OpenNebula::BackupJobPool.new(@client, user_flag)
+    end
+
+    def format_resource(bj, options = {})
+        bj_hash = bj.to_hash
+
+        str='%-15s: %-20s'
+        str_h1='%-80s'
+
+        # ----------------------------------------------------------------------
+        CLIHelper.print_header(
+            str_h1 % "BACKUP JOB #{bj['ID']} INFORMATION"
+        )
+        # ----------------------------------------------------------------------
+        puts format(str, 'ID', bj.id.to_s)
+        puts format(str, 'NAME', bj.name)
+        puts format(str, 'USER', bj['UNAME'])
+        puts format(str, 'GROUP', bj['GNAME'])
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against an allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html